### PR TITLE
fix(gui-client): don't delete the log file we're currently writing to

### LINKED ISSUE #4764: bug(connlib): connlib doesn't write to new logfile after logs are cleared
#### Describe the bug

There's no log files in `C:\Users\User\AppData\Local\dev.firezone.client\data\logs`, when I open the Logs tab it shows 0 files and 0 MB, and when I export a zip file it's empty.

#### To Reproduce

Can't replicate it.

I had the filter set to `firezone_gui_client=debug,firezone_tunnel=info,phoenix_channel=info,connlib_shared=info,connlib_client_shared=info,boringtun=info,snownet=info,str0m=info,warn`

but that worked when I tried it again.

#### Expected behavior

Expected at least some logs since `firezone_gui_client=debug` and `warn` are in there

#### Screenshots / Logs

N/A

#### Platform (please complete the following information)

- _Component (i.e. macOS client / Linux client / Gateway / Admin portal):_ Windows GUI Client
- _Firezone Version (e.g. 1.0.0 or N/A):_ Version 1.0.0 (d656cd5)
- _OS and version: (e.g. Ubuntu 22.04 or N/A):_ Windows 11
- _Deployment method: (e.g. Docker / Systemd / App Store or N/A):_ MSI from draft release

#### Additional context

I set the filter to `debug` and restarted the app and it worked.

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #4764 

```[tasklist]
- [x] Update changelog draft
- [x] Manual test Linux
```

### CODE REVIEW

File: rust/gui-client/src-tauri/src/client/logging.rs [hunk 1]
```diff
@@ -123,23 +124,65 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
...
+            if let Err(e) = tokio::fs::remove_file(path).await {
+                result = Err(e);
             }
         }
     }
 
     Ok(result?)
 }
 
+fn choose_logs_to_delete(paths: &[PathBuf]) -> Vec<&Path> {
```
Comment (thomaseizinger) [inline rust/gui-client/src-tauri/src/client/logging.rs:142]: Perhaps the following algorithm would be easier to understand:

- Given a file extension and a path
- List all files in a directory into a `Vec` (also filtering by `connlib.`)
- Sort using `.sort`
- Pop-off the last one
- Return list

Run call this fn twice, once for `.log` and once for `.jsonl`

**Review by thomaseizinger — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Alternatively, it should be possible to obtain the log file we are writing to from the logger handle.

File: rust/gui-client/src-tauri/src/client/logging.rs [hunk 2]
```diff
@@ -123,23 +124,65 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
...
+    let mut most_recent_stem = None;
+    for path in paths {
+        if path.extension() != Some(OsStr::new("log")) {
+            continue;
+        }
+        let stem = path
+            .file_stem()
+            .expect("Every file in the log dir should have a stem");
```
Comment (conectado) [inline rust/gui-client/src-tauri/src/client/logging.rs]: I'd rather not make an strong assumption about the path but simply `continue` if this fails.

File: rust/gui-client/src-tauri/src/client/logging.rs [hunk 3]
```diff
@@ -123,23 +124,65 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
...
+    }
+    let Some(most_recent_stem) = most_recent_stem else {
+        tracing::warn!(
+            "Nothing to delete, should be impossible since both processes always write logs"
+        );
+        return vec![];
+    };
+    let most_recent_stem = most_recent_stem
+        .to_str()
+        .expect("Every file in the log dir should have a UTF-8 file name");
```
Comment (conectado) [inline rust/gui-client/src-tauri/src/client/logging.rs]: Same here and all the `expect`s below

File: rust/gui-client/src-tauri/src/client/logging.rs [hunk 4]
```diff
@@ -123,23 +124,65 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
...
+
+    paths
+        .iter()
+        .filter(|path| {
+            let stem = path
+                .file_stem()
+                .expect("Every file in the log dir should have a stem");
+            let stem = stem
+                .to_str()
+                .expect("Every file in the log dir should have a UTF-8 file name");
```
Comment (jamilbk) [inline rust/gui-client/src-tauri/src/client/logging.rs]: Is this expect necessary? We've hit [issues](https://github.com/firezone/firezone/pull/5628) in the past regarding UTF-8 assumptions.

**Comment by ReactorScram** [comment]
> Alternatively, it should be possible to obtain the log file we are writing to from the logger handle.

Yeah that would be better long-term, especially once #5453 closes.

File: rust/gui-client/src-tauri/src/client/logging.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/logging.rs:142]: I know it's premature optimization, but I was thinking this runs in linear time whereas a sort will be super-linear

**Pull request merged**